File: 24-03-2024/Insert an Element at the Bottom of a Stack.js
//{ Driver Code Starts
//Initial Template for javascript


'use strict';


process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let [n,x] = readLine().trim().split(" ").map((x) => Number(x));
        let arr = readLine().trim().split(" ").map((x) => Number(x));
        let obj = new Solution();
        let res = obj.insertAtBottom(arr, x);
        let s = "";
        for(let it of res){
            s+=it+" ";
        }
        console.log(s);
    }
}

// } Driver Code Ends


//User function Template for javascript
/**
 * @param {Stack} st
 * @param {number} x
 * @returns {Stack}
*/

class Solution {
    //Function to insert an element at the bottom of a stack.
    insertAtBottom(st, x) {
        //your code Start
              if (st.length === 0) {
            // If the stack is empty, simply push the element.
            st.push(x);
        } else {
            // Remove all elements and store them in a function call stack.
            let temp = st.pop();
            // Recursively insert the element at the bottom.
            this.insertAtBottom(st, x);
            // Push the elements back onto the stack.
            st.push(temp);
        }
        return st;
      //Your Code End
    }
}

File: 24-03-2024/Insert an Element at the Bottom of a Stack.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG {
    static File file;
    static BufferedReader in;
    static PrintWriter out;

    public static void main(String args[]) throws IOException {
        in=new BufferedReader(new InputStreamReader(System.in));
        out=new PrintWriter(System.out);
        int t = Integer.parseInt(in.readLine());

        while (t-- > 0) {

            String s[] = in.readLine().trim().split(" ");
            int n = Integer.parseInt(s[0]);
            int x = Integer.parseInt(s[1]);
            s = in.readLine().trim().split(" ");
            Stack<Integer> st = new Stack<>();
            for (int i = 0; i < n; i++) {
                st.push(Integer.parseInt(s[i]));
            }
            Solution ob = new Solution();
            Stack<Integer> S = ob.insertAtBottom(st, x);
            for (int i : S) {
                out.print(i + " ");
            }
            out.println();
        }
        out.close();
    }
}

// } Driver Code Ends


//User function Template for Java
class Solution {
    public Stack<Integer> insertAtBottom(Stack<Integer> st, int x) {
      //Your Code Start
        if (st.isEmpty()) {
            // If the stack is empty, simply push the element
            st.push(x);
        } else {
            // Pop all elements and store them in the function call stack
            int temp = st.pop();
            // Recursively insert the element at the bottom
            insertAtBottom(st, x);
            // Push the elements back onto the stack
            st.push(temp);
        }
        return st;
      //Your Code End
        
    }
}
